contracts/SBCDepositContract.sol

```diff
@@ -335,12 +337,17 @@ contract SBCDepositContract is
         public
         failedWithdrawalProcessNonReentrant
     {
+        // Cache storage
+        mapping(uint256 => FailedWithdrawalRecord) storage withdrawals = failedWithdrawals;
+        uint256 pointer = failedWithdrawalsPointer;
+        uint256 failedCount = numberOfFailedWithdrawals;
+
         for (uint256 i = 0; i < _maxNumberOfFailedWithdrawalsToProcess; ++i) {
-            if (failedWithdrawalsPointer == numberOfFailedWithdrawals) {
+            if (pointer == failedCount) {
                 break;
             }
 
-            FailedWithdrawalRecord storage failedWithdrawalRecord = failedWithdrawals[failedWithdrawalsPointer];
+            FailedWithdrawalRecord memory failedWithdrawalRecord = withdrawals[pointer];
```

Comment:
This is a bug -- because of `memory` here in line `failedWithdrawalRecord.processed = true;` storage will not be changed.

Reply:
Nice catch, in `processFailedWithdrawal()` I differentiated between the cached and direct storage variables for this.

Sounds like this would be a good candidate for a new test as this bug passes all current tests.